Library/Homebrew/software_spec.rb

```diff
@@ -53,7 +53,7 @@ def owner=(owner)
     @resource.owner = self
     resources.each_value do |r|
       r.owner     = self
-      r.version ||= version
+      r.version ||= version.dup
```

Comment:
Is there any scenario where it does make sense that resources reference the same `Version` (or `HeadVersion`) instance as the formula itself and where changes to one of them should influence all the others? If not, I'm strongly in favor of `dup`.